code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -302,8 +310,96 @@
 			piece.item_flags |=  (STOPPRESSUREDAMAGE|AIRTIGHT)
 	update_icon(1)
 
-/obj/item/weapon/rig/process()
+/obj/item/weapon/rig/ui_action_click()
+	toggle_cooling(usr)
+
+/obj/item/weapon/rig/proc/toggle_cooling(var/mob/user)
+	if(cooling_on)
+		turn_cooling_off()
+	else
+		turn_cooling_on()
+
+/obj/item/weapon/rig/proc/turn_cooling_on()
+	if(!cell)
+		return
+	if(cell.charge <= 0)
+		usr << "<span class='notice'>\The [src] has no power!.</span>"
+		return
+	if(!suit_is_deployed())
+		usr << "<span class='notice'>The hardsuit needs to be deployed first!.</span>"
+		return
+
+	cooling_on = 1
+	usr << "<span class='notice'>You switch \the [src]'s cooling system on.</span>"
+
+
+/obj/item/weapon/rig/proc/turn_cooling_off(var/failed)
+	if(failed) visible_message("\The [src]'s cooling system clicks and whines as it powers down.")
+	else usr << "<span class='notice'>You switch \the [src]'s cooling system off.</span>"
+	cooling_on = 0
 
+/obj/item/weapon/rig/proc/get_environment_temperature()
+	if (ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		if(istype(H.loc, /obj/mecha))
+			var/obj/mecha/M = H.loc
+			return M.return_temperature()
+		else if(istype(H.loc, /obj/machinery/atmospherics/unary/cryo_cell))
+			return H.loc:air_contents.temperature

```

Comment:
Colons should be avoided whenever possible, and I'm fairly sure this will compile just fine without it.  You can add an if statement if you're paranoid as well.